streams/src/test/java/org/apache/kafka/streams/processor/internals/assignment/RackAwareTaskAssignorTest.java

```diff
@@ -0,0 +1,264 @@
+
+    private final MockInternalTopicManager mockInternalTopicManager = new MockInternalTopicManager(
+            time,
+            streamsConfig,
+            mockClientSupplier.restoreConsumer,
+            false
+    );
+
+    private Map<String, Object> configProps() {
+        final Map<String, Object> configurationMap = new HashMap<>();
+        configurationMap.put(StreamsConfig.APPLICATION_ID_CONFIG, APPLICATION_ID);
+        configurationMap.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, USER_END_POINT);
+
+        final ReferenceContainer referenceContainer = new ReferenceContainer();
+        /*
+        referenceContainer.mainConsumer = consumer;
+        referenceContainer.adminClient = adminClient;
+        referenceContainer.taskManager = taskManager;
+        referenceContainer.streamsMetadataState = streamsMetadataState;
+        referenceContainer.time = time;
+        */
+        configurationMap.put(InternalConfig.REFERENCE_CONTAINER_PARTITION_ASSIGNOR, referenceContainer);
+        return configurationMap;
+    }
+
+    @Test
+    public void disableActiveSinceMissingClusterInfo() {
+        final Cluster metadata = new Cluster(
+            "cluster",
+            new HashSet<>(Arrays.asList(node0, node1, node2)),
+            new HashSet<>(Arrays.asList(partitionInfo00, partitionInfo01)),
+            Collections.emptySet(),
+            Collections.emptySet()
+        );
+
+        final Map<UUID, Map<String, Optional<String>>> processRacks = new HashMap<>();
+
+        processRacks.put(process0UUID , Collections.singletonMap("consumer1", Optional.of("rack1")));
+
+        final RackAwareTaskAssignor assignor = new RackAwareTaskAssignor(
+            metadata,
+            Collections.singletonMap(new TaskId(1, 1), new HashSet<>(Arrays.asList(tp00, tp10))),
+            Collections.singletonMap(subtopology1, Collections.singleton(new TaskId(1, 1))),
+            processRacks,
+            mockInternalTopicManager,
+            new AssignorConfiguration(streamsConfig.originals()).assignmentConfigs()
+        );
+
+        // False since tp10 is missing in cluster metadata
+        assertFalse(assignor.canEnableForActive());
+    }
+
+    @Test
+    public void disableActiveSinceRackMissingInClient() {
+        final Cluster metadata = new Cluster(
+            "cluster",
+            new HashSet<>(Arrays.asList(node0, node1, node2)),
+            new HashSet<>(Arrays.asList(partitionInfo00, partitionInfo01)),
+            Collections.emptySet(),
+            Collections.emptySet()
+        );
+
+        final Map<UUID, Map<String, Optional<String>>> processRacks = new HashMap<>();
+
+        // Missing rackId config in client
+        processRacks.put(process0UUID , Collections.singletonMap("consumer1", Optional.empty()));
+
+        final RackAwareTaskAssignor assignor = new RackAwareTaskAssignor(
+            metadata,
+            Collections.singletonMap(new TaskId(1, 1), Collections.singleton(tp00)),
```

Comment:
Should we read from both "good" partitions?

Reply:
This is just to simplify to one task

Reply:
Not sure if I understand? We do have one task, but it could also read from more than one partitions?

My comments are mainly about readability of the test code -- we test different but very similar scenarios, but we setup each test very differently. To me, it seem easier to have some invariant in the code, and say: basic setup is X for all test, and each case is changing a single (or minimum) thing (not multiple/unnecessary things) to describe a single test scenario.

When I was reading the test code, I needed to re-read it multiple time until I understand what the test is actually doing because the setup change too many (and unnecessary) things at the same time, and it was unclear what changes are actually the required ones.

Reply:
Will extract common code to method to make it clear what the differences are